docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+
+2. For the image from the "docker.io/my_private_registry" registry, there must exist at least one signature that can be verified by the keys in keyData.
+
+3. For the image from the "registry.access.redhat.com", there must exist at least one signature that can be verified by the keys in `/etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release`.
+
+4. Reject all other images.
+
+The above example provides a secure and robust policy configuration, but the policy file still provides sufficient configuration flexibility for weak security requirements. For example, if you need to completely disable the container image signature verification function, policy file can be:
+
+```json
+{
+  "default": [
+    {
+      "type": "insecureAcceptAnything"
+    }
+  ],
+}
+```
+
+We must remind that this configuration is very insecure. It will pull all unsigned container images. Or if you want to unconditionally accept an image from a specific registry, you just need to change the policy of the member matching the registry under the corresponding scope to `insecureAcceptAnything`.
+
+### sigstore config
+
+The configuration file that specifies the container image sigstore is specified as all `.yaml` files in /etc/containers/registries.d directory. After the policy.json allowlist check is completed, the entries in the sigstore configuration file will claim where the container image is located, and the signature claim file of the container image will be downloaded from the specified URL (a webserver or a local directory). The sample sigstore configuration file matching the sample policy.json file in the above is as follows:
+
+```yaml
+default:
+    sigstore: file:///var/lib/containers/sigstore
+docker:
+  docker.io/my_private_registry:
+    sigstore: https://my-sigstore.com/example/sigstore
+  registry.access.redhat.com:
+    sigstore: https://access.redhat.com/webassets/docker/content/sigstore
+```
+
+If there is no `.yaml` file in the /etc/containers/registries.d/ directory when image-rs want to read it, as the way to obtain policy.json, image-rs should call the GetResource grpc service provided by the attestation-agent. After remote attestation, it is obtained from the user controlled relying KBS through the encrypted channel.
+
+### Image signature verification
+
+After completing the white list check and signature file download, signature verification is divided into the following steps:
+
+1. Parse the signature claim file and separate the JSON payload and cryptographic signature.
+
+2. Read the policy.json's "signedBy" policy, get the public key used to verify the signature from the "keyPath" or "keyData" field.
+
+3. Use the public key to verify the cryptographic signature in the signature claim file, and then compare whether the manifest digest in the JSON payload is consistent with that of the container image and whether the docker reference in the JSON payload is consistent with that of the container image.
```

Comment:
I would split the second half as a separate step for simplicity, and then split that in two:

4. Compare whether:
  * the manifest digest in the JSON payload is consistent with that of the container image
  * the docker reference in the JSON payload is consistent with that of the container image.